dcos_test_utils/gce.py

```diff
@@ -0,0 +1,254 @@
+    @wraps(f)
+    def handle_exception(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except HttpError as e:
+            if e.resp.status == 404:
+                log.exception("The resource you are trying to access doesn't exist")
+            elif e.resp.status == 409:
+                log.exception('''The specified resources exist and might be under an active operation
+                                   (operation conflict)''')
+            raise e
+
+    return handle_exception
+
+
+class GceWrapper:
+    @catch_http_exceptions
+    def __init__(self, credentials_dict):
+        credentials = GoogleCredentials.get_application_default()
+        self.compute = discovery.build('compute', 'v1', credentials=credentials)
+        self.deployment_manager = discovery.build('deploymentmanager', 'v2', credentials=credentials)
+        self.project_id = credentials_dict['project_id']
+
+    @catch_http_exceptions
+    def create_deployment(self, deployment_name, resources: str):
+        body = {
+            'name': deployment_name,
+            'target': {
+                'config': {
+                    'content': resources
+                }
+            }
+        }
+
+        response = self.deployment_manager.deployments().insert(project=self.project_id, body=body).execute()
+        log.debug('GceWrapper: create_deployment response: ' + str(response))
+
+    @catch_http_exceptions
+    def get_resources_info(self, name):
+        request = self.deployment_manager.resources().list(project=self.project_id, deployment=name)
+        while request is not None:
+            response = request.execute()
+            log.debug('GceWrapper: get_resources_info response: ' + str(response))
+
+            for resource_info in response['resources']:
+                yield resource_info
+
+            request = self.deployment_manager.resources().list_next(previous_request=request,
+                                                                    previous_response=response)
+
+    @catch_http_exceptions
+    def get_instance_info(self, name, zone):
+        response = self.compute.instances().get(project=self.project_id, zone=zone, instance=name).execute()
+        log.debug('GceWrapper: get_instance_info response: ' + str(response))
+        return response
+
+    @catch_http_exceptions
+    def get_deployment_info(self, name) -> dict:
+        """ Returns the dictionary representation of a GCE deployment resource. For details on the contents of this
+            resource, see https://cloud.google.com/deployment-manager/docs/reference/latest/deployments#resource"""
+        response = self.deployment_manager.deployments().get(project=self.project_id,
+                                                             deployment=name).execute()
+        log.debug('GceWrapper: get_deployment_info response: ' + str(response))
+        return response
+
+    @catch_http_exceptions
+    def delete_deployment(self, name):
```

Comment:
Why bother having a delete_deployment method here and a deployment.delete method? I think this should just live in Deployment